# Radio Control teleoperation configuration
# 
# The Graupner Falcon 12 receiver may be configured to 
# use either a pulse width modulated (PWM) mode or a  
# wireless serial mode SUMD. Our preference is to use SUMD
# which provides a 12 channels digital signal compared to 
# the 6 channel analog signal from PWM.
# 

num_channels: 12

# Set the RC channel and limits for linear velocity.
#
# Parameters
# ---------- 
# channel : int
#   RC channel. Valid values are 1 - 4.
# max_velocity : float
#   max linear velocity.
linear:
    x:
        channel: 3  # Assigned to left vertical stick. Ch1 when using PWM, Ch3 when using SUMD
        max_velocity: 0.37

# Set the RC channel and limits for angular velocity.
#
# Parameters
# ---------- 
# channel : int
#   RC channel. Valid values are 1 - 4.
# max_velocity : float
#   max angular velocity.
angular:
    z:
        channel: 1  # Assigned to right horizontal stick. Ch2 when using PWM, Ch1 when using SUMD
        max_velocity: 1.45

# Toggle the RC teleoperation on/off. 
# 
# Parameters
# ----------
# channel : int
#   RC channel assigned to switch. Valid values are 5 - 12.
disable_teleop:
    channel: 5  # Assigned to switch S3. Ch6 when using PWM, Ch5 when using SUMD

